src/extlibs/hrvo/hrvo_test.cpp

```diff
@@ -173,86 +211,192 @@ class HRVOTest : public ::testing::Test
-        std::cout << "Average time per tick = " << total_time.count() / frame
-                  << std::endl;
+        std::cout << "Avg time per tick = " << total_time.count() / frame << std::endl;
 
         output_file.close();
         std::cout << "Test information outputted to " << output_file_loc << std::endl;
     }
 };
 
-TEST_F(HRVOTest, div_b_edge_test)
+TEST_F(HRVOTest, stationary_friendly_robot_dodging_moving_friendly_robot)
 {
-    const Vector2 goal_offset  = Vector2(8.f, 0);
-    const Vector2 robot_offset = Vector2(0.f, -ROBOT_RADIUS * 2.5f);
-    for (std::size_t i = 0; i < 1; ++i)
-    {
-        const Vector2 position = -1 * (goal_offset / 2) + Vector2(0.f, 2.5f) +
-                                 (static_cast<float>(i) * robot_offset);
-        simulator.addAgent(position, simulator.addGoal(position + goal_offset));
-    }
-    add_static_obstacle(Vector2(0, 2.f), 0.75f);
-    create_div_b_field();
+    std::vector<std::pair<Point, Point>> friendly_start_dest_points = {
+        std::pair(Point(0.0, 0.0), Point(0.0, 0.0)),
+        std::pair(Point(-2.0, 0.0), Point(4.0, 0.0))};
+    instantiate_robots_in_world(friendly_start_dest_points, {});
+}
+
+TEST_F(HRVOTest, stationary_friendly_robot_dodging_moving_enemy_robot)
+{
+    std::vector<std::pair<Point, Point>> friendly_start_dest_points = {
+        std::pair(Point(0.0, 0.0), Point(0.0, 0.0))};
+    std::vector<std::pair<Point, Vector>> enemy_position_velocity_pairs = {
+        std::pair(Point(-2.0, 0.0), Vector(1.0, 0.0))};
+    instantiate_robots_in_world(friendly_start_dest_points,
+                                enemy_position_velocity_pairs);
+    setSimulationTimeout(3.f);
+}
+
+TEST_F(HRVOTest, friendly_and_enemy_robot_moving_towards_each_other)
+{
+    std::vector<std::pair<Point, Point>> friendly_start_dest_points = {
+        std::pair(Point(5.0, 0.0), Point(-5.0, 0.0))};
+    std::vector<std::pair<Point, Vector>> enemy_position_velocity_pairs = {
+        std::pair(Point(-2.0, 0.0), Vector(1.0, 0.0))};
+    instantiate_robots_in_world(friendly_start_dest_points,
+                                enemy_position_velocity_pairs);
+    setSimulationTimeout(5.f);
+}
+
+TEST_F(HRVOTest, multiple_friendly_robots_lining_up)
+{
+    std::vector<std::pair<Point, Point>> friendly_start_dest_points = {
+        std::pair(Point(5.0, 0.0), Point(-3.0, 0.5)),
+        std::pair(Point(5.0, -0.9), Point(-3.0, 0.7)),
+        std::pair(Point(5.0, -0.6), Point(-3.0, 0.9)),
+        std::pair(Point(5.0, -0.3), Point(-3.0, 1.1))};
+    instantiate_robots_in_world(friendly_start_dest_points, {});
+}
+
+TEST_F(HRVOTest, single_friendly_robot_moving_in_line)
+{
+    std::vector<std::pair<Point, Point>> friendly_start_dest_points = {
+        std::pair(Point(-5.0, 0.0), Point(5.0, 0.0))};
+    instantiate_robots_in_world(friendly_start_dest_points, {});
+}
+
+TEST_F(HRVOTest, destination_between_friendly_robot_and_stationary_enemy_robot)
```

Comment:
can we plot the goal position too?

Reply:
I can look into it.

Reply:
I've added it. This was a good idea, its nice to be able to visualize the goal position. Makes it easier to see the oscillation aswell.